pydatalab/tests/server/test_blocks.py

```diff
@@ -0,0 +1,180 @@
+def test_invalid_block_type(admin_client, default_sample_dict):
+    """Test that invalid block types are rejected."""
+    sample_id = "test_sample_invalid_block"
+    sample_data = default_sample_dict.copy()
+    sample_data["item_id"] = sample_id
+
+    response = admin_client.post("/new-sample/", json=sample_data)
+    assert response.status_code == 201
+
+    response = admin_client.post(
+        "/add-data-block/",
+        json={
+            "block_type": "nonexistent_block_type",
+            "item_id": sample_id,
+            "index": 0,
+        },
+    )
+
+    assert response.status_code == 400
+    assert "Invalid block type" in response.json["message"]
+
+
+@pytest.mark.dependency(depends=["test_get_all_available_block_types"])
+def test_add_multiple_blocks_to_sample(admin_client, default_sample_dict):
+    """Test adding multiple different block types to the same sample."""
+    sample_id = "test_sample_multiple_blocks"
+    sample_data = default_sample_dict.copy()
+    sample_data["item_id"] = sample_id
+
+    response = admin_client.post("/new-sample/", json=sample_data)
+    assert response.status_code == 201
+
+    block_types_to_test = list(BLOCK_TYPES.keys())[:3]
+    block_ids = []
+
+    for i, block_type in enumerate(block_types_to_test):
+        response = admin_client.post(
+            "/add-data-block/",
+            json={
+                "block_type": block_type,
+                "item_id": sample_id,
+                "index": i,
+            },
+        )
+
+        assert response.status_code == 200, f"Failed to add {block_type} block at index {i}"
+        assert response.json["status"] == "success"
+        block_ids.append(response.json["new_block_obj"]["block_id"])
+
+    response = admin_client.get(f"/get-item-data/{sample_id}")
+    assert response.status_code == 200
+
+    item_data = response.json["item_data"]
+    assert len(item_data["blocks_obj"]) == len(block_types_to_test)
+    assert len(item_data["display_order"]) == len(block_types_to_test)
+
+    added_block_types = [block["blocktype"] for block in item_data["blocks_obj"].values()]
+    for block_type in block_types_to_test:
+        assert block_type in added_block_types
+
+
+def test_block_permissions(client, admin_client, default_sample_dict):
+    """Test that normal users can add blocks to samples they have access to."""
+    sample_id = "test_sample_user_permissions"
+    sample_data = default_sample_dict.copy()
+    sample_data["item_id"] = sample_id
+
+    response = admin_client.post("/new-sample/", json=sample_data)
+    assert response.status_code == 201
+
+    block_type = list(BLOCK_TYPES.keys())[0]
+    response = client.post(
+        "/add-data-block/",
+        json={
+            "block_type": block_type,
+            "item_id": sample_id,
+            "index": 0,
+        },
+    )
+
+    if response.status_code == 200:
+        assert response.json["status"] == "success"
+    elif response.status_code == 401:
+        pass
```

Comment:
I think this test will always just pass through with 401 right? Could we instead test that the "normal" client can make a sample, add a block, then the admin can add another block, but an unauthenticated client cannot?